Tests/LoggingTests/TestLogger.swift

```diff
@@ -23,7 +23,21 @@ internal struct TestLogging {
     private let recorder = Recorder() // shared among loggers
 
     func make(label: String) -> LogHandler {
-        return TestLogHandler(label: label, config: self.config, recorder: self.recorder)
+        return TestLogHandler(
+            label: label,
+            config: self.config,
+            recorder: self.recorder,
+            metadataProvider: LoggingSystem.metadataProvider
+        )
+    }
+
+    func makeWithMetadataProvider(label: String, metadataProvider: Logger.MetadataProvider) -> LogHandler {
```

Comment:
collapse this into `make` and make `metadataProvider` optional with default of `.none`?

Reply:
That'll cause a series of ambiguity errors AFAIR, I'll give it a shot tho.

Reply:
Yeah I'll push back on this one, it causes a lot more pain in our test code than necessary; 
We can't do `LoggingSystem.bootstrapInternal(logging.makeWithMetadataProvider)` style anymore and would have to spell out all the closures to disambiguate.

Note that this is only our test code, so the verbose name is not a concern IMHO